addons/suggest_locations/models/suggest_by_product.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, _
+from itertools import groupby
+from .suggest_locations_policy import SuggestLocationPolicy, get_selection
+
+
+class ByProduct(SuggestLocationPolicy):
+    """Find locations based on other placed that product is stored"""
+
+    @classmethod
+    def name(cls):
+        return "by_product"
+
+    def get_values_from_mls(self, mls):
+        return {
+            "product": mls.product_id.ensure_one(),
+            "parent_location": mls.u_picking_type_id.default_location_dest_id.ensure_one(),
```

Comment:
No need for ensure_one

Reply:
Ignore it, in this case we need it, we can have `mls` with different picking types.